homeassistant/components/fronius/__init__.py

```diff
@@ -168,6 +177,56 @@ async def _create_solar_net_device(self) -> DeviceInfo:
         )
         return solar_net_device
 
+    async def _init_devices_inverter(self) -> None:
+        """Retrieve inverter information from host."""
+        _inverter_infos: list[FroniusDeviceInfo] = []
+        try:
+            _inverter_infos = await self._get_inverter_infos()
+        except ConfigEntryNotReady as err:
+            raise err
+        except ConnectionError as err:
+            # Might occur during a re-scan. Attempt again on next scheduled scan
+            _LOGGER.error("Scan of %s failed. %s", self.host, err)
+
+        _LOGGER.debug("Processing inverters for: %s", _inverter_infos)
+        for _inverter_info in _inverter_infos:
+            _inverter_name = (
+                f"{DOMAIN}_inverter_{_inverter_info.solar_net_id}_{self.host}"
+            )
+
+            # Add found inverter only not already existing
+            if (
+                next(
+                    (
+                        inv
+                        for inv in self.inverter_coordinators
+                        if inv.name == _inverter_name
+                    ),
+                    None,
+                )
+                is None
+            ):
```

Comment:
I think this should be checked by `unique_id` or `solarnet_id` instead of name for better readability. The name prefix doesn't really matter for uniqueness, and it's suffix (host) is unique to the config entry anyway. 

Also consider checking rewriting this as a guard with `continue` and unindent the following block. 
```suggestion
# pseudo code
            if _inverter_info.solarnet_id in [inv.solarnet_id for inv in self.inverter_coordinators:
                continue
```